### PR TITLE
Allow timestamp to be set by the file writing utility functions

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/update-done/update-done.c [hunk 1]
```diff
@@ -31,64 +28,23 @@
...
-                       "TIMESTAMP_NSEC=" NSEC_FMT "\n",
-                       timespec_load_nsec(ts));
+        if (asprintf(&message,
+                     MESSAGE
+                     "TIMESTAMP_NSEC=" NSEC_FMT "\n",
+                     timespec_load_nsec(ts)) < 0)
+                return log_oom();
 
-        r = fflush_and_check(f);
+        r = write_string_file_atomic_label_ts(path, message, ts);
```
Comment (poettering) [inline src/update-done/update-done.c:45]: this previously handled EROFS errors special and just debug logged about them. Do you knowingly drop that?

Comment (keszybz) [reply src/update-done/update-done.c:45]: Good catch. Updated.

**Comment by poettering** [comment]
lgtm

**Comment by poettering** [comment]
ci failure appears unrelated

**Pull request merged**